### DISCUSSION TITLE
Use of extensions requires connections to US military contractor (GitHub) to download executable code.

### DISCUSSION THREAD

**Discussion by sneak** [standardnotes/forum#2702] [General]
I have GitHub firewalled at my router.  None of my themes work now.

I subscribe to Standard Notes for privacy and security.  I find it unacceptable that it is using a military contractor and active partner in the US surveillance state to a) leak my IP/location, and b) download executable code.

This gives Microsoft (GitHub's owner) and the US military by extension the ability to subvert Standard Notes' encryption and compromise my data. I never signed up for that.

**Comment by atmoio** [comment]
As of today, we do have a dependence on GitHub for many of our operations. While our plan is generally to reduce dependencies over time, I can't yet say when we will be 100% GitHub free for our CDN. I'd say it's on the mid-term horizon.

**Comment by sneak** [comment]
https://github.com/standardnotes/forum/issues/1017

**Comment by cyb3rz3us** [comment]
I think the more salient question here is this --- Is there any credibility to the suggestion that using GitHub for one's open source projects in any way exposes that software and\or its users to surveillance or other malicious activity by the US military?

My gut tell me No but I would like to hear from others...

**Comment by sneak** [comment]
There is also the more mundane point that when extensions are enabled and GitHub (and the sync service) are inaccessible due to being offline that the client will lose data.  I ran into this issue, despite the sync service being accessible through my firewall.

**Comment by cyb3rz3us** [comment]
_"...GitHub (and the sync service) are inaccessible due to being offline that the client will lose data..."_

Perhaps...but a totally separate point from the one being discussed here...

**Comment by JaspalSuri** [comment]
> There is also the more mundane point that when extensions are enabled and GitHub (and the sync service) are inaccessible due to being offline that the client will lose data. I ran into this issue, despite the sync service being accessible through my firewall.

Hi @sneak, please look at Mo's reply here: https://github.com/standardnotes/desktop/issues/567#issuecomment-646113605
